session.go

```diff
@@ -283,6 +288,24 @@ func (s *Session) init() error {
 	return nil
 }
 
+func (s *Session) waitForConnAvailable() error {
+	select {
+	case <-s.connAva:
+	case <-time.After(s.cfg.Timeout):
```

Comment:
Waiting for just `s.cfg.Timeout` is not correct, the connection initialization simply takes different amount of time. There is `s.cfg.ConnectTimeout`, but that is used just for dialing. When establishing a connection, there is retry policy. We also send some data back and forth, like query to set keyspace, so the connection takes more time to setup. In case the connection fails, we need to wait until we stop retrying.

If you did not change the `policyHostPool.addHost` to non-blocking, you would be able to handle all this waiting within `session.init` and without need for selecting a specific timeout. I believe it's beter to have the waiting implementation encapsulated just within that method. (See also #1369, where the waiting is all implemented within `session.init`)

Reply:
Done.